mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewModel.kt

```diff
@@ -416,18 +417,22 @@ abstract class MavericksViewModel<S : MavericksState>(
         deliveryMode: DeliveryMode,
         action: suspend (T) -> Unit
     ): Job {
-        val flow = if (lifecycleOwner == null || MavericksTestOverrides.FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
-            this
-        } else if (deliveryMode is UniqueOnly) {
-            val lastDeliveredValue: T? = lastDeliveredValue(deliveryMode)
-            this
-                .assertOneActiveSubscription(lifecycleOwner, deliveryMode)
-                .dropWhile { it == lastDeliveredValue }
-                .flowWhenStarted(lifecycleOwner)
-                .distinctUntilChanged()
-                .onEach { lastDeliveredStates[deliveryMode.subscriptionId] = it }
-        } else {
-            flowWhenStarted(lifecycleOwner)
+        val flow = when {
+            lifecycleOwner == null || MavericksTestOverrides.FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER -> this
+            deliveryMode is UniqueOnly -> configureAndFilter(lifecycleOwner, deliveryMode.subscriptionId, true, deliveryMode) { flow, _ ->
+                flow.dropWhile { it == lastDeliveredValue(deliveryMode.subscriptionId) }
```

Comment:
is there a preexisting bug where an null value will not get delivered, even if no value was previously delivered, because `null` also represents not present in the map?

Reply:
yea, i think that would be the case. If initial value is null and it is never changed, uniqueOnly would not deliver that initial value